### PR TITLE
Improve the error handling for updating try pushes after landing.

### PR DESCRIPTION
**Pull request by jgraham**
Don't abort early in case we have a complete try push, and ensure that
in the case of error we always add a comment to the bug so that it's
clear something has gone wrong.

### CODE REVIEW

File: sync/command.py [hunk 1]
```diff
@@ -303,13 +303,7 @@ def update_landing():
...
                     update_landing()
                     if current_landing.latest_try_push == try_push:
-                        landing.try_push_complete(git_gecko,
-                                                  git_wpt,
-                                                  try_push,
-                                                  current_landing,
-                                                  allow_push=kwargs["push"],
-                                                  accept_failures=accept_failures,
-                                                  tasks=tasks)
+                        landing.push_to_gecko(git_gecko, git_wpt, current_landing)
```
Comment (dhdavvie) [inline sync/command.py]: I'm not sure why we would skip `try_push_complete` here, maybe I'm missing something?

Comment (jgraham) [reply sync/command.py]: If the try push is marked as complete we must have already run the `try_push_complete` logic on that try push without hitting an error. So we can immediately do the `push_to_gecko` step without rerunning the whole function on the try push.

File: sync/landing.py [hunk 2]
```diff
@@ -1023,17 +1007,58 @@ def try_push_complete(git_gecko, git_wpt, try_push, sync, allow_push=True,
...
 
+def needinfo_users():
+    needinfo_users = [item.strip() for item in
+                      (env.config["gecko"]["needinfo"]
+                       .get("landing", "")
+                       .split(","))]
+    return [item for item in needinfo_users if item]
+
+
+def record_failure(sync, log_msg, bug_msg, fixup_msg=None):
```
[the remark this answers is not in the document]
Comment (jgraham) [reply sync/landing.py]: Perhaps, although they're all handled errors, and the subsequent `AbortError` will show up in newrelic.

**Comment by jgraham** [comment]
@dhdavvie This is updated now FYI

**Pull request merged**